File: 2346-largest-3-same-digit-number-in-string/2346-largest-3-same-digit-number-in-string.py
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        s=""
        for i in range(1,len(num)-1):
            if(num[i-1]==num[i] and num[i+1]==num[i]):
                s=max(s,num[i]*3)
                
        return s
        

File: 2346-largest-3-same-digit-number-in-string/2346-largest-3-same-digit-number-in-string.cpp
class Solution {
public:
    string largestGoodInteger(string num) {
    int maxnum = -1;
    string output="";
        for (int i = 1; i < num.length()-1; i++) {
        if(num[i-1]==num[i] && num[i+1]==num[i]){
                string s = string(3,num[i]);
                int k = stoi(s); 
                if (k > maxnum) {
                    maxnum = k;
                    output=s;
                }}}
    return maxnum == -1 ? "" : output;
        
    }
};